x/claim/types/mission_test.go

```diff
@@ -0,0 +1,50 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/spn/testutil/sample"
+	claim "github.com/tendermint/spn/x/claim/types"
+)
+
+func TestMission_Validate(t *testing.T) {
+	invalidWeightGreaterThanOne := sample.Mission(r)
+	amt, err := sdk.NewDecFromStr("1.0000001")
+	require.NoError(t, err)
+	invalidWeightGreaterThanOne.Weight = amt
```

Comment:
I think it would be more visual and in the idea of table test to have the full definition of mission in the case table below

For handling error from `NewDecFromStr` we could implement a `Dec` method in `testconstructor` package like we have `Shares` and `Coins`